experimenter/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/FormFeatureValue/validators.ts

```diff
@@ -347,26 +348,32 @@ export function schemaAutocomplete(schema: Record<string, unknown>) {
   };
 }
 
-/**
- * Parse from FML errors to Diagnostics
- */
-function parseFmlErrors(view: TestableEditorView): Diagnostic[] {
-  const diagnostics: Diagnostic[] = [];
-
-  const doc = view.state.doc;
-  const text = doc.toString();
-  const from = text.indexOf("error");
-  if (from === -1) return diagnostics;
-
-  const to = from + "error".length;
-
-  diagnostics.push(createDiagnostic(from, to, "oh no!"));
-  return diagnostics;
-}
+export function fmlLinter(
+  slug: string,
+  feature: getConfig_nimbusConfig_allFeatureConfigs,
+) {
+  return async function (view: TestableEditorView) {
+    const url = `/api/v5/fml-errors/${slug}/`;
+    const response = await fetch(url, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        featureSlug: feature.slug,
+        featureValue: view.state.doc.toString(),
+      }),
+    });
+    const errors: Record<string, any>[] = await response.json();
+
+    return errors.map((e: Record<string, any>) => {
+      const position = documentPosition(
+        view.state.doc,
+        e["line"] + 1,
+        e["col"],
+      );
 
-export function fmlLinter() {
-  return function (view: TestableEditorView) {
-    // TODO call endpoint to retrieve errors, remove temp error highlighting
-    return parseFmlErrors(view);
+      return createDiagnostic(position, position, e["message"]);
```

Comment:
I think this also returns the `token` in `e` so you can add `e['token'].length` to the second position column

Reply:
Oh it's called 'highlight' but yes exactly! Much better:

<img width="885" alt="image" src="https://github.com/mozilla/experimenter/assets/119884/7b69399f-4627-49da-9fe1-34f488a9c9b6">